subprojects/logging/src/integTest/groovy/org/gradle/internal/logging/console/taskgrouping/RichVerboseConsoleTypeFunctionalTest.groovy

```diff
@@ -44,22 +48,84 @@ class RichVerboseConsoleTypeFunctionalTest extends AbstractConsoleGroupedTaskFun
                 dependsOn helloWorld, byeWorld, silence
             }
         """
-    }
-
-    @Unroll
-    def "can have verbose task output according to --console"() {
         when:
-        executer.withConsole(mode)
         succeeds('all')
 
         then:
-        result.groupedOutput.task(':helloWorld').output == HELLO_WORLD_MESSAGE
-        result.groupedOutput.task(':byeWorld').output == BYE_WORLD_MESSAGE
+        result.groupedOutput.task(':helloWorld').output == 'Hello world'
+        result.groupedOutput.task(':byeWorld').output == 'Bye world'
         hasSilenceTaskOutput == result.groupedOutput.hasTask(':silence')
 
         where:
         mode    | hasSilenceTaskOutput
         Rich    | false
         Verbose | true
     }
+
+    def "long running task's output are divided by other output"() {
+        given:
+        buildFile << '''
+def sleepFor(String desc, int times){
+    (1..times).each {
+         Thread.sleep(1000)
+         println "${desc} wake up ${it}"
+    }
+}
+
+task('longRunning').doLast {
+    sleepFor('longRunning', 6)
```

Comment:
The sleep value is a bit magical, use [`GroupingProgressLogEventGenerator .LONG_RUNNING_TASK_OUTPUT_FLUSH_TIMEOUT `](https://github.com/gradle/gradle/blob/master/subprojects/logging/src/main/java/org/gradle/internal/logging/sink/GroupingProgressLogEventGenerator.java#L50).